transifex/native/core.py

```diff
@@ -105,26 +105,35 @@ def translate(
 
     def get_translation(self, source_string, language_code, _context,
                         is_source=False):
-        """ Try to retrieve the translation.
+        """Return the proper translation for the given string
+        and language code.
 
-            A translation is a serialized source_string with ICU format
-            support, e.g.
-            '{num, plural, one {Ένα τραπέζι} other {{num} τραπέζια}}'
+        A translation is a serialized source_string with ICU format
+        support, e.g.
+        '{num, plural, one {Ένα τραπέζι} other {{num} τραπέζια}}'
+
+        Supports strings in the source language as well, which means
+        that if there is a translation available in the cache
+        for the source language, it will be used instead of the
+        original source_string provided here.
         """
+        pluralized, plurals = parse_plurals(source_string)
+        key = generate_key(string=source_string, context=_context)
+        translation_template = self._cache.get(key, language_code)
+        if (translation_template is not None and pluralized and
+                translation_template.startswith('{???')):
+            variable_name = source_string[1:source_string.index(',')].strip()
+            translation_template = '{{{var}{content}'.format(
+                var=variable_name,
+                content=translation_template[4:],
+            )
 
-        if is_source:
+        # If rendering the source language and there is no
+        # (overridden) translation in the cache, use the original
+        # source string
+        if is_source and not translation_template:
```

Comment:
Why not just `if not translation_template:`?

Also with that if in place, what happens if we are in target language and the string is not in the cache? Does the ErrorPolicy trigger?

Reply:
The reason we only want to do this for the source language is that in all other cases we want to delegate the rendering to the missing policy. The missing policy doesn't make sense for source strings, as the fallback should always be the original source strings.

This method here returns the found translation (`translation_template`), which is then used by the `[translate()](https://github.com/transifex/transifex-python/blob/20be85e64adf887e0eb5c88dec2a38e0e5356d6c/transifex/native/core.py#L100-L104)` method. So the logic is the following:
- if source and no translation in cache, return the original source
- if source and translation in cache, return the cached translation
- in any other case, return anything the cache has, even if empty

The final decisions are made by [StringRenderer.render()](https://github.com/transifex/transifex-python/blob/20be85e64adf887e0eb5c88dec2a38e0e5356d6c/transifex/native/rendering.py#L38).